amida-tech/amida-messaging-microservice
# Feature/filter threads by role

If this PR fixes a bug, you _must_ add test cases representative of the bug.
#### What's this PR do?
- Restricts the create-thread and reply-to-thread endpoints to the user scopes specified in the `MESSAGING_SERVICE_THREAD_SCOPES` env variable
#### Related JIRA tickets:
#### How should this be manually tested?

- Update your env and test.env from the example env
   - `MESSAGING_SERVICE_THREAD_SCOPES=["admin", "clinician"]`
- Given a clinician user and a non-clinician user
- Create a thread with the clinician user (this should succeed)
- Create a thread with the non-clinician user (this should fail)
- Reply to a thread with the clinician user (this should succeed)
- Reply to a thread with the non-clinician user (this should fail)

- Run the tests and make sure everything passes (make sure your env.test is updated before doing this)

#### Any background context you want to provide?
#### Screenshots (if appropriate):
